src/arch/src/x86_64/mptable.rs

```diff
@@ -215,7 +215,7 @@ pub fn setup_mptable(mem: &GuestMemoryMmap, num_cpus: u8) -> Result<()> {
         checksum = checksum.wrapping_add(compute_checksum(&mpc_ioapic.0));
     }
     // Per kvm_setup_default_irq_routing() in kernel
-    for i in 0..16 {
+    for i in 0..24 {
```

Comment:
Maybe it would be better to quit hardcoding this value?

Reply:
+1.
```suggestion
    for i in 0..IRQ_MAX + 1 {
```